### PR TITLE
Adds pseudo-transaction models

### LINKED ISSUE #201: Pseudo-transaction models
To be able to instantiate arbitrary transactions from the ledger, you need to also be able to recognize and handle [pseudo-transactions](https://xrpl.org/pseudo-transaction-types.html).

We should have models for these and a way to instantiate "any transaction you may have found in a ledger" into the appropriate model. That way people could monitor all ledger transactions, and interact with them in a Pythonic way.

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds a PseudoTransaction base model, each of the 3 pseudo-transaction models, and tests for each models. It also adds some bugfixes that were uncovered when testing. It also moves `TransactionType` and `PseudoTransactionType` to its own folder in `xrpl.models.transactions`, to avoid circular import problems.

### Context of Change

Python library 1.1.0

Resolves #201 

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Test Plan

Added tests that convert to/from XRPL JSON format (since people can't send these to the ledger and will only receive them from the ledger).

### CODE REVIEW

File: xrpl/models/transactions/pseudo_transactions/enable_amendment.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    transaction_type: PseudoTransactionType = field(
+        default=PseudoTransactionType.ENABLE_AMENDMENT,
+        init=False,
+    )
+
+    #: The Flags value of the EnableAmendment pseudo-transaction indicates the status
+    #: of the amendment at the time of the ledger including the pseudo-transaction.
+    #: A Flags value of 0 (no flags) or an omitted Flags field indicates that the
+    #: amendment has been enabled, and applies to all ledgers afterward.
+    flags: int = 0
```
Comment (ledhed2222) [inline xrpl/models/transactions/pseudo_transactions/enable_amendment.py:66]: this should be inherited from the base `Transaction`, no?

**Review by ledhed2222 — COMMENTED** [review]
one question - otherwise excellent work!

File: xrpl/models/transactions/pseudo_transactions/enable_amendment.py (hunk 1 above)
Comment (mvadari) [reply xrpl/models/transactions/pseudo_transactions/enable_amendment.py:66]: Yeah, it is, but I wanted to override the docs.

Comment (ledhed2222) [reply xrpl/models/transactions/pseudo_transactions/enable_amendment.py:66]: should be a `Union[int, List[int]]`

**Review by ledhed2222 — APPROVED** [review]

**Pull request merged**